src/java/main/org/apache/zookeeper/server/quorum/Leader.java

```diff
@@ -1376,4 +1382,8 @@ public static String getPacketType(int packetType) {
             return "UNKNOWN";
         }
     }
+
+    private boolean isRunning() {
+        return self.isRunning() && zk.isRunning();
```

Comment:
Is it really logical AND here or should we be using logical OR? I'm wondering if there is any chance that just one of those `isRunning` calls evaluates to true.